solidity/contracts/token/libs/MovableCollateralRouter.sol

```diff
@@ -113,20 +117,45 @@ abstract contract MovableCollateralRouter is FungibleTokenRouter {
         uint256 amount,
         ITokenBridge bridge
     ) external payable onlyRebalancer onlyAllowedBridge(domain, bridge) {
-        address rebalancer = _msgSender();
+        bytes32 recipient = _recipient(domain);
+
+        Quote[] memory quotes = bridge.quoteTransferRemote(
+            domain,
+            recipient,
+            amount
+        );
+
+        uint256 collateralFee = 0;
+        for (uint256 i = 0; i < quotes.length; i++) {
+            if (quotes[i].token == token()) {
+                collateralFee += quotes[i].amount;
```

Comment:
I think we can break here.

Reply:
if there are multiple quotes with the same token, wont that undercount?

Reply:
I don't think multiple quotes with the same token should be supported, maybe we can add that to the spec for `quoteTransferRemote`.